### PR TITLE
Added 'Swap Data and Metadata' option to 'View' menu

### PR DESCRIPTION
**Pull request by N-thony**
Swapping Metadata and Data windows. I suggest that when the options about Data View and Column data are checked in the view menu, the option Swap metadata and data window will appear in the View menu so that if you check, the swapping will occur. This solves issue #6208 
@rdstern this is ready for review. Thanks.

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony It seems to be working quite well, but not exactly as I was hoping.  I am happy with what is there though.
I was hoping for the "Swap Data and Metadata" entry to be available when I start.  If you click on it (at the start, it would replace the data (i.e. close that) by the metadata, i.e. open the metadata in its place.
Currently I can swap when I have 3 windows open.  I like that, but would also like the option with just the 2 windows.

**Review by rdstern — CHANGES_REQUESTED** [review]
Aha, I can do what I want with your current system, but it takes multiple steps.
I opened the Column Metadata - and now the swap appears.  Now I swap.  Now I close the data.  
I would like those 3 steps from the start, by haing the Swap menu item visible all the time.  I just click on that and it swaps them over.

**Comment by N-thony** [comment]
@rdstern  Thanks for your comment above. I think I resolved it. Please could you review?

**Comment by lloyddewit** [comment]
@Ivanluv please could you review? thanks

**Comment by N-thony** [comment]
@lloyddewit sorry, before @Ivanluv to review this, following the chat with @rdstern there is a small change I made in the code that I would like @rdstern to test first then @Ivanluv will proceed with the review. @rdstern please could you test the small change I made? Thanks

**Comment by N-thony** [comment]
@lloyddewit sorry, before @Ivanluv to review this, following the chat with @rdstern there is a small change I made in the code that I would like @rdstern to test first then @Ivanluv will proceed with the review. @rdstern please could you test the small change I made? Thanks

**Comment by N-thony** [comment]
@rdstern I have added some combinations for the swapping system. Please could you test the change? Thanks

**Review by rdstern — APPROVED** [review]
@N-thony this is now ok from my point of view

**Comment by lloyddewit** [comment]
@Ivanluv this PR is ready again for your review, thanks

**Comment by Ivanluv** [comment]
@lloyddewit  am looking at this now

**Comment by Ivanluv** [comment]
@lloyddewit  this looks okay

**Comment by shadrackkibet** [comment]
I tested this and seems to work - no code review. However, I have one comment.  I did the following.

- Loaded data
- Swap Data and Metadata
- Open Data view via **View>Data View**
- Close Data view via **View>Data View**

I thought because **Swap data and metadata** is checked the layout should remain swapped i.e Column Metadata side by side with Output window.

File: instat/frmMain.vb [hunk 1]
```diff
@@ -452,6 +453,18 @@ Public Class frmMain
...
+                splMetadata.Panel1.Controls.Add(ucrDataViewer)
+            ElseIf mnuViewSwapDataAndMetadata.Checked AndAlso mnuViewDataView.Checked AndAlso Not mnuViewColumnMetadata.Checked Then
+                splDataOutput.Panel1.Controls.Add(ucrColumnMeta)
+                splMetadata.Panel1.Controls.Add(ucrDataViewer)
+                mnuViewColumnMetadata.Checked = True
+                mnuViewDataView.Checked = False
+            Else
+                splDataOutput.Panel1.Controls.Add(ucrDataViewer)
+                splMetadata.Panel1.Controls.Add(ucrColumnMeta)
+            End If
```
Comment (lloyddewit) [inline instat/frmMain.vb:467]: restructure to remove duplication?
```suggestion
            If mnuViewSwapDataAndMetadata.Checked AndAlso mnuViewDataView.Checked Then
                splDataOutput.Panel1.Controls.Add(ucrColumnMeta)
                splMetadata.Panel1.Controls.Add(ucrDataViewer)
                If Not mnuViewColumnMetadata.Checked Then
                    mnuViewColumnMetadata.Checked = True
                    mnuViewDataView.Checked = False
                End If
            Else
                splDataOutput.Panel1.Controls.Add(ucrDataViewer)
                splMetadata.Panel1.Controls.Add(ucrColumnMeta)
            End If
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
@lloyddewit thanks for the comment above. I resolved it.

**Review by lloyddewit — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern You approved on 10 Feb. Since then, there was a small code change that should not affect the functionality. Please could you approve again? thanks

**Review by rdstern — APPROVED** [review]

**Comment by lloyddewit** [comment]
@N-thony Please can you resolve the merge conflicts? We can then merge, thanks

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern The only change since your last approval were changes to fix merge conflcts in frmMain. Please could you approve again? thanks

**Comment by shadrackkibet** [comment]
> I tested this and seems to work - no code review. However, I have one comment. I did the following.
> 
> * Loaded data
> * Swap Data and Metadata
> * Open Data view via **View>Data View**
> * Close Data view via **View>Data View**
> 
> I thought because **Swap data and metadata** is checked the layout should remain swapped i.e Column Metadata side by side with Output window.

Just before this is approved and merged. I would like to check if this was considered.

**Comment by rdstern** [comment]
@shadrack you are quite right and I hadn't tested that combination.  It is a useful one and I agree with you.
But @N-thony this is ok and has been going on a long time.  So, if easy to fix, then please do.  Otherwise I can approve and this can be a further small issue.

Just to clarify what Shadrack found.  
a) View > Swap data and metadata - looks fine but you don't see the data - just the metadata.  And View > data View is correctly not ticked.
b) View > Data View

Now you see the data and usefully in a wide Window right across.  That's quite a useful layout with the 3 windows

c) Now View > Data View, because you have finished looking at the data and want to continue as before, i.e. with the meat-data alongside the Output.  But it isn't like that.

**Review by rdstern — COMMENTED** [review]
@N-thony the recent problem still seems to be there.  I hope it can be fixed soon.

**Comment by lloyddewit** [comment]
> 
> 
> @shadrack you are quite right and I hadn't tested that combination. It is a useful one and I agree with you.
> But @N-thony this is ok and has been going on a long time. So, if easy to fix, then please do. Otherwise I can approve and this can be a further small issue.

@rdstern - @N-thony wasn't sure how to fix this and he asked my view.
To solve this, I think we would have to update `frmMain.UpdateLayout()` and possibly also look at the event handlers for `mnuViewDataView`, `mnuViewColumnMetaData` and `mnuViewSwapDataAndMetadata`.  This is significant work (>1 person-day including extra testing?). I assume this is not a priority for the sprint. I suggest that we make a small issue instead.

**Review by rdstern — DISMISSED** [review]
This is not worth that amount of work.  The changers are excellent, and this is a very slight inconvenience, i.e. you have to press another couple of buttons to get what is needed.
So I have no hesitation in considering this a "feature" and therefore am happy to approve.

**Comment by lloyddewit** [comment]
@N-thony Please could you 
- make an issue for this last unresolved 'feature'
- resolve the merge conflicts

thanks

**Comment by N-thony** [comment]
> @N-thony Please could you
> 
> * make an issue for this last unresolved 'feature'
> * resolve the merge conflicts
> 
> thanks

@lloyddewit thanks, I resolved the conflicts. I will write an issue for the last unresolved 'feature'.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@N-thony Another PR completed, thank you :)

@rdstern @maxwellfundi @shadrackkibet I did a quick regression test and it looks OK. Please could one of you approve so we can merge? 
Thanks

**Review by rdstern — APPROVED** [review]
Great

**Pull request merged**